File: frontend/src/types/interview.ts
export interface Interview {
  id: string;
  application_id: string;
  user_id: string;
  interview_type: string;
  status: string;
  scheduled_at: string;
  duration_minutes: number;
  location: string | null;
  interviewer_name: string | null;
  notes: string | null;
  feedback: string | null;
  result: string | null;
  created_at: string;
  updated_at: string;
  company_name: string | null;
  job_title: string | null;
}

export interface InterviewCreate {
  application_id: string;
  interview_type: string;
  scheduled_at: string;
  duration_minutes?: number;
  location?: string;
  interviewer_name?: string;
  notes?: string;
}

export interface InterviewUpdate {
  interview_type?: string;
  status?: string;
  scheduled_at?: string;
  duration_minutes?: number;
  location?: string;
  interviewer_name?: string;
  notes?: string;
  feedback?: string;
  result?: string;
}

File: backend/app/schemas/interview.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class InterviewCreate(BaseModel):
    application_id: UUID
    interview_type: str
    scheduled_at: datetime
    duration_minutes: int = 60
    location: Optional[str] = None
    interviewer_name: Optional[str] = None
    notes: Optional[str] = None


class InterviewUpdate(BaseModel):
    interview_type: Optional[str] = None
    status: Optional[str] = None
    scheduled_at: Optional[datetime] = None
    duration_minutes: Optional[int] = None
    location: Optional[str] = None
    interviewer_name: Optional[str] = None
    notes: Optional[str] = None
    feedback: Optional[str] = None
    result: Optional[str] = None


class InterviewResponse(BaseModel):
    id: UUID
    application_id: UUID
    user_id: UUID
    interview_type: str
    status: str
    scheduled_at: datetime
    duration_minutes: int
    location: Optional[str] = None
    interviewer_name: Optional[str] = None
    notes: Optional[str] = None
    feedback: Optional[str] = None
    result: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    company_name: Optional[str] = None
    job_title: Optional[str] = None

    model_config = {"from_attributes": True}
